app/controller/pyproject.toml

```diff
@@ -10,7 +10,7 @@ chainsail-controller = 'chainsail.controller.run:run'
 chainsail-controller-local = 'chainsail.controller.run_local:run'
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = "^3.8,<3.11"
```

Comment:
This subtle change is necessary to make `controller` compatible with `chainsail-helpers` (as it has a similar <3.11 constraint).